src/client/dfuse/dfuse.h

```diff
@@ -82,12 +82,20 @@ struct dfuse_eq {
 	pthread_t           de_thread;
 
 	struct d_slab_type *de_read_slab;
+	struct d_slab_type *de_pre_read_slab;
 	struct d_slab_type *de_write_slab;
 };
 
-/* Maximum size dfuse expects for read requests, this is not a limit but rather what is expected */
+/* Maximum size dfuse expects for read requests, this is not a limit but rather what is expected
+ * This is the maximum size expected from the kernel, increasing this without changing kernel
+ * behavior will have no effect.
+ */
 #define DFUSE_MAX_READ (1024 * 1024)
 
+/* Maximum file-size for pre-read requests.  This can be any value at the cost of
+ * memory consumption */
+#define DFUSE_MAX_PRE_READ (1024 * 1024 * 4)
```

Comment:
it would be great to make that configurable

Reply:
I don't really want to make this a per-container setting although it could be, that leave command line of probably environment variable as the best options.  I think it's going to be about tuning a sensible default rather than recommending users change the settings so I think an environment variable would probably be best here.